### ISSUE TITLE
Softmax in SDR classifier can overflow for high learning rates

### ISSUE THREAD

**Issue by natoromano** [numenta/nupic#3174]
For high learning rates, computing the softmax scores can overflow in the SDR classifier, at the exponentiation step:

```
# softmax normalization
expOutputActivation = numpy.exp(outputActivation)
```

The common practice to fix this is to offset all the activations by their maximum value, but that can have an effect of performance when the number of buckets is high.

NB: To reproduce this, a learning rate of 0.5 is enough in `nupic.integration.opf.opf_experiments_results_test.OPFExperimentResultsTest`.

**Comment by natoromano** [comment]
@ywcui1990 what do you think? Should I add this fix when implementing the C++ version?
